bindings/matrix-sdk-crypto-ffi/src/lib.rs

```diff
@@ -350,6 +353,94 @@ impl<T: Fn(i32, i32)> ProgressListener for T {
     }
 }
 
+/// An encryption algorithm to be used to encrypt messages sent to a room.
+pub enum EventEncryptionAlgorithm {
+    /// Olm version 1 using Curve25519, AES-256, and SHA-256.
+    OlmV1Curve25519AesSha2,
+    /// Megolm version 1 using AES-256 and SHA-256.
+    MegolmV1AesSha2,
+}
+
+impl From<EventEncryptionAlgorithm> for RustEventEncryptionAlgorithm {
+    fn from(a: EventEncryptionAlgorithm) -> Self {
+        match a {
+            EventEncryptionAlgorithm::OlmV1Curve25519AesSha2 => {
+                RustEventEncryptionAlgorithm::OlmV1Curve25519AesSha2
+            }
+            EventEncryptionAlgorithm::MegolmV1AesSha2 => {
+                RustEventEncryptionAlgorithm::MegolmV1AesSha2
+            }
+        }
+    }
+}
+
+/// Who can see a room's history.
+pub enum HistoryVisibility {
+    /// Previous events are accessible to newly joined members from the point
+    /// they were invited onwards.
+    ///
+    /// Events stop being accessible when the member's state changes to
+    /// something other than *invite* or *join*.
+    Invited,
+
+    /// Previous events are accessible to newly joined members from the point
+    /// they joined the room onwards.
+    /// Events stop being accessible when the member's state changes to
+    /// something other than *join*.
+    Joined,
+
+    /// Previous events are always accessible to newly joined members.
+    ///
+    /// All events in the room are accessible, even those sent when the member
+    /// was not a part of the room.
+    Shared,
+
+    /// All events while this is the `HistoryVisibility` value may be shared by
+    /// any participating homeserver with anyone, regardless of whether they
+    /// have ever joined the room.
+    WorldReadable,
+}
+
+impl From<HistoryVisibility> for RustHistoryVisibility {
+    fn from(h: HistoryVisibility) -> Self {
+        match h {
+            HistoryVisibility::Invited => RustHistoryVisibility::Invited,
+            HistoryVisibility::Joined => RustHistoryVisibility::Joined,
+            HistoryVisibility::Shared => RustHistoryVisibility::Shared,
+            HistoryVisibility::WorldReadable => RustHistoryVisibility::Shared,
+        }
+    }
+}
+
+/// Settings for an encrypted room.
+///
+/// This determines the algorithm and rotation periods of a group session.
+pub struct EncryptionSettings {
+    /// The encryption algorithm that should be used in the room.
+    pub algorithm: EventEncryptionAlgorithm,
+    /// How long the session should be used before changing it. Time in seconds.
+    pub rotation_period: u64,
+    /// How many messages should be sent before changing the session.
+    pub rotation_period_msgs: u64,
+    /// The history visibility of the room when the session was created.
+    pub history_visibility: HistoryVisibility,
```

Comment:
I am not sure I understand where is the client supposed to get the `history_visibility` from, given that the session creation is opaque to the client, and `history_visibility` may change during the lifetime of a room (unlike fixed `rotation_period_msgs` or `rotation_period` parameters). How is this supposed to be used?

Reply:
It's the current history visibility of the room. Follow up doc PR https://github.com/matrix-org/matrix-rust-sdk/pull/1144

Reply:
Ah understand now, thanks for the clarification